src/test/java/org/folio/rest/impl/protection/InvoicesProtectionTest.java

```diff
@@ -126,4 +137,76 @@ public void testModifyUnitsList(ProtectedOperations operation) {
 
     validateNumberOfRequests(0, 0);
   }
+
+  @Test
+  @Parameters({
+    "CREATE",
+    "UPDATE"
+  })
+  public void testAssigningSoftDeletedUnit(ProtectedOperations operation) {
+    logger.info("=== Test invoice contains \"soft deleted\" unit id - expecting of call only to Units API ===");
+    
+    final Headers headers = prepareHeaders(X_OKAPI_TENANT, ALL_DESIRED_PERMISSIONS_HEADER, X_OKAPI_USER_ID);
+
+    // Prepare acq unit in storage for update case and
+    AcquisitionsUnit unit1 = prepareTestUnit(false);
+    AcquisitionsUnit unit2 = prepareTestUnit(true);
+    AcquisitionsUnit unit3 = prepareTestUnit(true);
+    // Add all acq units as mock data
+    addMockEntry(ACQUISITIONS_UNITS, unit1);
+    addMockEntry(ACQUISITIONS_UNITS, unit2);
+    addMockEntry(ACQUISITIONS_UNITS, unit3);
+
+    // Prepare invoice with 2 acq units (one is "soft deleted") and add it as mock data for update case
+    Invoice invoice = prepareInvoice(Arrays.asList(unit1.getId(), unit2.getId()));
+    
+    // Add the third unit to request
+    invoice.getAcqUnitIds().add(unit3.getId());
+    
+    Errors errors = operation.process(INVOICE_PATH, encodePrettily(invoice),
+        headers, APPLICATION_JSON, HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt()).as(Errors.class);
+    
+    assertThat(errors.getErrors(), hasSize(1));
+    Error error = errors.getErrors().get(0);
+    assertThat(error.getCode(), equalTo(ACQ_UNITS_NOT_FOUND.getCode()));
+    assertThat(error.getAdditionalProperties().get(ACQUISITIONS_UNIT_IDS), instanceOf(List.class));
+    
+    // Verify number of sub-requests
+    validateNumberOfRequests(1, 0);
+
+    List<String> ids = (List<String>) error.getAdditionalProperties().get(ACQUISITIONS_UNIT_IDS);
+    if (operation == UPDATE) {
+      assertThat(ids, contains(unit3.getId()));
+    } else {
+      assertThat(ids, containsInAnyOrder(unit2.getId(), unit3.getId()));
+    }
+  }
+
+  @Test
+  public void testGetInvoiceWithAssignedSoftDeletedUnit() {
+    logger.info("=== Test invoice contains only \"soft deleted\" unit id - expecting of call only to Units API ===");
+
+    final Headers headers = prepareHeaders(X_OKAPI_TENANT, ALL_DESIRED_PERMISSIONS_HEADER, X_OKAPI_USER_ID); 
+    // Prepare acq unit in storage for update case
+    AcquisitionsUnit unit1 = prepareTestUnit(true);
```

Comment:
Just to improve test, the unit can be created as protected for read i.e. 
```java
AcquisitionsUnit unit1 = prepareTestUnit(true).withProtectRead(true);
```
In this case if unit would not be "soft deleted", the `403` would be returned...
I should do this in original implementation for orders but thought about it only now :)